cmake/macros/SundialsAddTest.cmake

```diff
@@ -186,11 +186,11 @@ macro(SUNDIALS_ADD_TEST NAME EXECUTABLE)
                                              OR (SUNDIALS_TEST_MPIRUN_COMMAND)))
         if(SUNDIALS_TEST_MPIRUN_COMMAND)
           set(RUN_COMMAND
-              "${SUNDIALS_TEST_MPIRUN_COMMAND} ${MPIEXEC_NUMPROC_FLAG} ${SUNDIALS_ADD_TEST_MPI_NPROCS} ${MPIEXEC_PREFLAGS}"
+              "${SUNDIALS_TEST_MPIRUN_COMMAND} ${MPIEXEC_PREFLAGS} ${MPIEXEC_NUMPROC_FLAG} ${SUNDIALS_ADD_TEST_MPI_NPROCS} ${MPIEXEC_POSTFLAGS}"
```

Comment:
The [current version of the CMake documentation is clear](https://cmake.org/cmake/help/latest/module/FindMPI.html#usage-of-mpiexec).  

>${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${MPIEXEC_MAX_NUMPROCS}
  ${MPIEXEC_PREFLAGS} EXECUTABLE ${MPIEXEC_POSTFLAGS} ARGS

I am going to update our usage to match that.

Reply:
Ah, yes, putting preflags before the numprocs argument does not work with `flux` because the PREFLAGS for flux has to be set to `run` since the command is `flux run` (and CMake will error our if you set `MPIEXEC_EXECUTABLE to `flux run`).

Reply:
Ah, I was only looking at the variable descriptions and just needed to scroll down. On another project we do the following in our run script and it seems to work on Tioga 
```
mpiexec_executable=${MPIEXEC_EXECUTABLE:-"srun"}
# If using flux, append "run" after the flux executable path
if [[ "${mpiexec_executable}" == "flux" ]]
then
    mpiexec_executable="$(which ${mpiexec_executable}) run"
    flux jobs
    flux resource list
else
    mpiexec_executable="$(which ${mpiexec_executable})"
fi
```
where `MPIEXEC_EXECUTABLE` is an environment variable set by the job then in the `cmake` line we have
```
-DMPIEXEC_EXECUTABLE="${mpiexec_executable}" \
```
```

Reply:
OK I went with the solution from your other project.

Reply:
Oops, I was looking at the wrong job. The other project approach is causing a failure.